stocksite/stocks/views/buyView.py

```diff
@@ -0,0 +1,67 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from accounts.models import Account
+from transactions.models import Transaction
+from time import time
+
+
+class BuyView(APIView):
+    """
+    API endpoint that allows stocks to be bought.
+    """
+    def post(self, request):
+        # Get request data
+        username = request.data.get("username")
+        stockSymbol = request.data.get("stockSymbol")
+        amount = float(request.data.get("amount"))
+
+        # Find user account
+        account = Account.objects.filter(username=username).first()
+
+        # If account is non-existing, log errorEvent to Transaction
+        if account is None:
+            transaction = Transaction(
+                type='errorEvent',
+                timestamp=int(time()*1000),
```

Comment:
might need to store time variable in local variable too, in case there is a thread context switch.

Might not matter though. .